faiss/impl/ScalarQuantizer.cpp

```diff
@@ -1024,55 +1195,57 @@ struct DistanceComputerByte<Similarity, 8> : SQDistanceComputer {
  * specialization
  *******************************************************************/
 
-template <class Sim>
+template <class Sim, class Sim_default>
```

Comment:
I don't understand why it is crucial to introduce `class Sim_default` here. @naveentatikonda, could you please give some comments? Thanks.

Reply:
In ARM architecture, as we don't have the templates implemented for all the Quantizers(except QT_fp16) with SIMDWIDTH 8 it is throwing errors when we were trying to build it as it is not able to find the corresponding templates . So, I introduced a new argument `Sim_default` without messing up this switch case or adding more if else conditions. Pls let me know if there is any better way to do this.

Reply:
Agreed with @alexanderguzhva. 
Let's try to not make ScalarQuantizer even more complex than it already is.

Reply:
Thank you for the suggestions. Will implement the 2nd approach suggested by @alexanderguzhva